### PR TITLE
Add required anti-affinity for harvester network manager

### PR DESCRIPTION
**Pull request by yaocw2020**
related issue https://github.com/harvester/harvester/issues/1722

### CODE REVIEW

File: charts/harvester-network-controller/values.yaml [hunk 1]
```diff
@@ -32,7 +32,7 @@ affinity: {}
 
 # Default value for harvester-network-controller-manager deployment
 manager:
-  replicas: 2
+  replicas: 1
```
Comment (gitlawr) [inline charts/harvester-network-controller/values.yaml:35]: Why do we scale down to 1

**Comment by gitlawr** [comment]
Does it affect upgrades?

**Comment by guangbochen** [comment]
> Does it affect upgrades?

yes, added a workaround on https://github.com/harvester/docs/pull/67/commits/99529f3dac7a96d35c82db4980c34f2eace77c13, also better to fix the label issue early instead of later.

File: charts/harvester-network-controller/values.yaml (hunk 1 above)
Comment (guangbochen) [reply charts/harvester-network-controller/values.yaml:35]: we may leverage the Rancher scale feature of single-node case https://github.com/harvester/harvester/blob/0345caa981570fcad6e15401b33656f6ad5d2bda/deploy/charts/harvester/templates/deployment.yaml#L6-L24

File: charts/harvester-network-controller/templates/deployment.yaml [hunk 2]
```diff
@@ -4,17 +4,27 @@ kind: Deployment
 metadata:
   namespace: {{ .Release.Namespace }}
   labels:
-    {{- include "harvester-network-controller.labels" . | nindent 4 }}
+    {{- include "harvester-network-controller-manager.labels" . | nindent 4 }}
+  {{- if .Values.manager.replicas }}
+  # The annotation does not support 0 replicas, which is used in integration tests.
```
Comment (gitlawr) [inline charts/harvester-network-controller/templates/deployment.yaml:9]: This seems to be unnecessary as we don't set 0 replicas for network-controller in https://github.com/harvester/harvester/blob/master/tests/integration/runtime/construct.go#L50

Comment (gitlawr) [reply charts/harvester-network-controller/templates/deployment.yaml:9]: At least we should remove the integration tests comments

File: charts/harvester-network-controller/values.yaml (hunk 1 above)
Comment (gitlawr) [reply charts/harvester-network-controller/values.yaml:35]: We should keep the 2 replicas even with the annotation change

File: charts/harvester-network-controller/templates/deployment.yaml (hunk 2 above)
Comment (guangbochen) [reply charts/harvester-network-controller/templates/deployment.yaml:9]: in order to pass the CI u may need to config the default replicas to 1 with refer to https://github.com/harvester/charts/blob/master/charts/harvester-cloud-provider/ci/kind-values.yaml

**Review by gitlawr — APPROVED** [review]
LGTM

**Review by guangbochen — APPROVED** [review]

**Pull request merged**